tools/ui/src/candid.ts

```diff
@@ -92,15 +94,31 @@ export async function getCycles(canisterId: Principal): Promise<bigint|undefined
 
 export async function getNames(canisterId: Principal) {
   try {
-    const actor = getProfilerActor(canisterId);
-    const blob = await actor.__get_names() as number[];
-    const decoded = IDL.decode([IDL.Vec(IDL.Tuple(IDL.Nat16, IDL.Text))], Uint8Array.from(blob))[0] as Array<[number,string]>;
+    const paths : CanisterStatus.Path[] = [{
+      kind: 'metadata',
+      path: 'name',
+      key: 'name',
+      decodeStrategy: 'raw',
+    }];
+    const status = await CanisterStatus.request({ agent, canisterId, paths });
+    const blob = status.get('name') as ArrayBuffer;
+    const decoded = IDL.decode([IDL.Vec(IDL.Tuple(IDL.Nat16, IDL.Text))], blob)[0] as Array<[number,string]>;
     decoded.forEach(([id, name]) => names[id] = name);
   } catch(err) {
     return undefined;
   }
 }
 
+async function getDidJsFromMetadata(canisterId: Principal): Promise<undefined | string> {
+  const status = await CanisterStatus.request({ agent, canisterId, paths: ['candid'] });
```

Comment:
I think it's `candid:service` https://github.com/dfinity/sdk/blob/master/src/dfx/src/lib/metadata/names.rs

Reply:
`agent-js` has its own naming convention: https://github.com/dfinity/agent-js/blob/main/packages/agent/src/canisterStatus/index.ts#L181